enzyme/Enzyme/GradientUtils.h

```diff
@@ -71,7 +71,7 @@
 
 using namespace llvm;
 
-enum class DerivativeMode { Forward, Reverse, Both };
+enum class DerivativeMode { AugmentedForward, Forward, Reverse, Both };
```

Comment:
Need not be this PR but I may rename this so old Forward is unambiguously becomes "Augmented Forward Pass of reverse mode" and the new one is "Forward Mode AD"

Reply:
How about:
AugmentedForward -> Forward
Forward -> Passthrough

Reply:
Passthrough feels weird in that I wouldn't understand what is meant by it.

Maybe:
AugmentedForward => ForwardMode
Forward => ReverseModePrimal
Reverse => ReverseModeGradient
Both      => ReverseModeCombined

@vchuravy thoughts as well?

Reply:
shouldn't it be: AugmentedForward => ForwardModeDirectionalDerivative

Reply:
I don't think there's forward mode AD that isn't directional in this sense so it feel redundant?

Reply:
++ for the ReverseMode rename.

Reply:
done 5b45bf589e27fd83f9c586798aa38979e3317eef